lib/elasticsearch-cache.js

```diff
@@ -0,0 +1,206 @@
+const _ = require('lodash');
+const colors = require('colors');
+const moment = require('moment');
+const client = require('./elasticsearch')();
+const crypto = require('crypto');
+
+// ElasticSearch
+const index = 'cache';
+const type = 'cache';
+
+module.exports = class ElasticSearchCache {
+
+  constructor(options) {
+
+    this.options = _.defaults(options, {
+      es_client: client,
+      default_ttl: 3600
+    });
+  }
+
+  /**
+   * Retrieving cache document by key
+   *
+   * @param key - cache key (might be a string or object)
+   * @returns {Promise}
+   */
+  get(key) {
+    return new Promise((resolve, reject) => {
+
+      if (!key) {
+        let err = 'cache.get: Missed key parameter.';
+
+        console.error(err);
+        resolve({
+          ok: false,
+          err: err
+        });
+      }
+
+      if (typeof key === 'object') {
+        key = JSON.stringify(key, Object.getOwnPropertyNames(key));
+      }
+
+      let _id = crypto.createHash('sha1').update(key).digest('hex');
+
+      console.log('cache.get: Requesting cache [%s]', colors.green(key));
```

Comment:
@fqjony Let's use `debug`. Do not execute `console.log`. Especially in third party npm modules. It clutters up logs outputs/